### PR TITLE
Update crate documentation

### PR DESCRIPTION
**Pull request by rajarshimaitra**
### Description

After #99 the previous documentation have been removed and new docs as per `structopts` documentation. This PR adds more documentation across the crate.. 

This PR is above #102 , to accommodate all the further refactoring changes. 

The Readme About section have been updated with more details.. Readme format made aligned with the BDK project itself..

The Readme file is used itself as the crate level documentation in docs.rs too..   

#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk-cli/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,12 +1,56 @@
...
 
 ## About
 
-This project provides a command line interface (cli) Bitcoin wallet library and [`REPL`](https://en.wikipedia.org/wiki/Read%E2%80%93eval%E2%80%93print_loop)
-wallet tool based on the [bdk](https://github.com/bitcoindevkit/bdk) library.
+This project provides a command-line Bitcoin wallet application using the latest [BDK APIs](https://docs.rs/bdk/0.19.0/bdk/wallet/struct.Wallet.html). This is used mostly as an high level integration testing framework, by the BDK team, using the [tests](/tests/integration.rs) modules.
+
+But if you are planning to use BDK in your own wallet project, bdk-cli is a nice playground to get started with. It allows easy testnet and regtest wallet operations, to try out whats possible with descriptors, miniscripts, and BDK APIs. For more information on BDK refer the [website](https://bitcoindevkit.org/), and the [rust docs](https://docs.rs/bdk/latest/bdk/index.html)
+
+bdk-cli can be compiled in various different ways to suit the experimental needs.
```
Comment (waterst0ne) [inline README.md:34]: bdk-cli can be compiled in various different ways to suit the experimental needs.

File: README.md [hunk 2]
```diff
@@ -1,12 +1,56 @@
...
+</div>
 
-![CI](https://github.com/bitcoindevkit/bdk-cli/workflows/CI/badge.svg)
-![Code Coverage](https://github.com/bitcoindevkit/bdk-cli/workflows/Code%20Coverage/badge.svg)
 
 ## About
 
-This project provides a command line interface (cli) Bitcoin wallet library and [`REPL`](https://en.wikipedia.org/wiki/Read%E2%80%93eval%E2%80%93print_loop)
-wallet tool based on the [bdk](https://github.com/bitcoindevkit/bdk) library.
+This project provides a command-line Bitcoin wallet application using the latest [BDK APIs](https://docs.rs/bdk/0.19.0/bdk/wallet/struct.Wallet.html). This is used mostly as an high level integration testing framework, by the BDK team, using the [tests](/tests/integration.rs) modules.
```
Comment (waterst0ne) [inline README.md:30]:
```suggestion
This project provides a command-line Bitcoin wallet application using the latest [BDK APIs](https://docs.rs/bdk/0.19.0/bdk/wallet/struct.Wallet.html). This interface is used mostly as a framework for high level integration testing, by the BDK team, using the [tests](/tests/integration.rs) modules.
```

File: README.md [hunk 3]
```diff
@@ -1,12 +1,56 @@
...
+
+But if you are planning to use BDK in your own wallet project, bdk-cli is a nice playground to get started with. It allows easy testnet and regtest wallet operations, to try out whats possible with descriptors, miniscripts, and BDK APIs. For more information on BDK refer the [website](https://bitcoindevkit.org/), and the [rust docs](https://docs.rs/bdk/latest/bdk/index.html)
```
Comment (waterst0ne) [inline README.md:32]:
```suggestion
If you consider using BDK in your own wallet project, bdk-cli is a nice playground to get started with. It allows easy testnet and regtest wallet operations, to try out whats possible with descriptors, miniscripts, and BDK APIs. For more information on BDK refer the [website](https://bitcoindevkit.org/), and the [rust docs](https://docs.rs/bdk/latest/bdk/index.html)
```

File: README.md (hunk 1 above)
Comment (waterst0ne) [inline README.md:34]:
```suggestion
bdk-cli can be compiled with different features to suit the experimental needs.
```

File: README.md [hunk 4]
```diff
@@ -1,12 +1,56 @@
...
+     - `electrum` : Connects the wallet to an electrum server.
+     - `compact_filters` : Deploy a BIP157 node to get blockchain data from bitcoin p2p network.
+     - `esplora-ureq/reqwest` : Connects the wallet to a esplora server sync/asynchronously.
+  - Extra Utility Tools
+     - `repl` : use bdk-cli as a [REPL](https://codewith.mu/en/tutorials/1.0/repl) shell (useful for quick hand testing wallet operations).
+     - `compiler` : opens up bdk-cli policy compiler commands.
+     - `verify` : uses `bitcoinconsensus` to verify transactions at every `sync` call of the wallet.
+     - `reserves` : opens up bdk-cli **Proof of Reserves** operation commands using the [bdk-reserves plugin](https://github.com/weareseba/bdk-reserves). (requires `electrum`)
+   - Automated Node Backend
+     - `regtest-bitcoin` : Auto deploys a regtest bitcoin node, connects the wallet, and exposes core rpc commands via `bdk-cli node` subsommands.
```
Comment (waterst0ne) [inline README.md:49]:
```suggestion
     - `regtest-bitcoin` : Auto deploys a regtest bitcoin node, connects the wallet, and exposes core rpc commands via `bdk-cli node` subcommands.
```

File: README.md [hunk 5]
```diff
@@ -1,12 +1,56 @@
...
+     - `regtest-electrum` : Auto deploys a `electrsd` connected to a `bitcoind` and an wallet connectded to `electrsd`. `bdk-cli node` subcommand still calls the `bitcoind` RPC.
```
Comment (waterst0ne) [inline README.md:50]:
```suggestion
     - `regtest-electrum` : Auto deploys a `electrsd` connected to a `bitcoind` and a wallet connectded to `electrsd`. `bdk-cli node` subcommand still calls the `bitcoind` RPC.
```

**Comment by rajarshimaitra** [comment]
Thanks for nits @waterst0ne .. Really helpful..

**Comment by rajarshimaitra** [comment]
@waterst0ne Updated with your comments..

**Comment by rajarshimaitra** [comment]
Rebased on latest #102

**Comment by rajarshimaitra** [comment]
Rebased on top of #102 ..

**Comment by rajarshimaitra** [comment]
Thanks @notmandatory for the 102 merge.. Rebased on master..
 
This is basic documentation update and should be straight forward to review..

**Review by notmandatory — APPROVED** [review]
ACK f8a5999289fd0c648b6c18b7f0011339ecaa736d

**Comment by notmandatory** [comment]
@rajarshimaitra I added a commit with some small grammar and punctuation changes. For the punctuation I primarily added periods to the end of lines in the rust docs, which is how they do it in other rust docs I've seen.  If these changes look OK feel free to merge the PR, or I'm happy to merge it tomorrow with your OK.

**Comment by rajarshimaitra** [comment]
Thanks @notmandatory for the review.. All looks good to me.. Merged..

**Pull request merged**